content/compose/environment-variables/set-variables.md

```diff
@@ -48,6 +32,57 @@ services:
     image: 'webapp:v1.5'
 ```
 
+Inserting values in this way is known as interpolation.
+
+## Interpolation syntax
+
+Interpolation is applied for unquoted and double-quoted values.
+Both braced (`${VAR}`) and unbraced (`$VAR`) expressions are supported.
+
+For braced expressions, the following formats are supported:
+- Direct substitution
+  - `${VAR}` -> value of `VAR`
+- Default value
+  - `${VAR:-default}` -> value of `VAR` if set and non-empty, otherwise `default`
+  - `${VAR-default}` -> value of `VAR` if set, otherwise `default`
+- Required value
+  - `${VAR:?error}` -> value of `VAR` if set and non-empty, otherwise exit with error
+  - `${VAR?error}` -> value of `VAR` if set, otherwise exit with error
+- Alternative value
+  - `${VAR:+replacement}` -> `replacement` if `VAR` is set and non-empty, otherwise empty
+  - `${VAR+replacement}` -> `replacement` if `VAR` is set, otherwise empty
+
+For more information, see [Interpolation](../compose-file/12-interpolation.md) in the Compose Specification. 
+
+## Ways to set variables with interpolation
+
+### `.env` file
+
+An `.env` file in Docker Compose is a text file used to define  variables that should be made available to Docker containers when running `docker compose up`. This file typically contains key-value pairs of  variables, and it allows you to centralize and manage configuration in one place. The `.env` file is useful if you have multiple variables you need to store.
+
+The `.env` file is the default method for setting variables. The `.env` file should be placed at the root of the project directory next to your `compose.yaml` file. For more information on formatting an environment file, see [Syntax for environment files](env-file.md).
+
+Basic example: 
+
+```console
+$ cat .env
+## define COMPOSE_DEBUG based on DEV_MODE, defaults to false
+COMPOSE_DEBUG=${DEV_MODE:-false}
+
+$ cat compose.yaml 
+  services:
+    webapp:
+      image: my-webapp-image
+      environment:
+        - DEBUG=${COMPOSE_DEBUG}
+
+$ DEV_MODE=true docker compose config
+services:
+  webapp:
+    environment:
+      DEBUG: "true"
+```
+
 #### Additional information 
```

Comment:
This section should be moved away from this page as this isn't related to variables in compose.yaml ("interpolation") but container's environement